lib/ui/compositing/scene.cc

```diff
@@ -87,6 +90,44 @@ Dart_Handle Scene::toImage(uint32_t width,
                                             height, raw_image_callback);
 }
 
+static sk_sp<DlImage> CreateDeferredImage(
+    bool impeller,
+    std::shared_ptr<LayerTree> layer_tree,
+    uint32_t width,
+    uint32_t height,
+    fml::WeakPtr<SnapshotDelegate> snapshot_delegate,
+    fml::RefPtr<fml::TaskRunner> raster_task_runner,
+    fml::RefPtr<SkiaUnrefQueue> unref_queue) {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (impeller) {
```

Comment:
Here and else where, Can we put the logic in DlDeferredGPUImpeller ?

Reply:
I didn't do that because it created some weirdness about dependencies but let me see if I can find an appropriate target for this type to live in so we can do that.

Reply:
The existing dependencies are ui->impeller->display_list

If we put `display_list_deferred_image_gpu_impeller.h` and `display_list_deferred_image_gpu_impeller.cc` in the `ui `instead of the `impeller` (like `display_list_deferred_image_gpu_skia` does), it seems to fix this problem.

Please let me know if i missed something.

Reply:
Ok, I've moved these to `DlDeferredImageGPUImpeller::Make`, and moved those TUs to `//flutter/lib/ui/painting` next to the skia implementations.